actions/getProfileData.ts

```diff
@@ -0,0 +1,45 @@
+import { getMumbleUserByIdentifier } from '@/utils/helpers/users/getMumbleUserByIdentifier';
+import { getAllFollowers } from '@/utils/helpers/followers/getFollowers';
+import { IMumbleUser } from '@/utils/interfaces/mumbleUsers.interface';
+import { IMumbleFollowers } from '@/utils/interfaces/mumbleFollowers.interface';
+import { getAllFollowees } from '@/utils/helpers/followers/getFollowees';
+
+export interface IGetProfileDataArgs {
+  profileIdentifier: string;
+  loggedInUserIdentifier?: string;
+}
+
+export interface IProfileData {
+  profileData: IMumbleUser;
+  profileFollowers: IMumbleFollowers[];
+  profileFollowees: IMumbleFollowers[];
+  loggedUserFollowers: IMumbleFollowers[];
+  loggedUserFollowees: IMumbleFollowers[];
+}
+
+export const getProfileData = async ({
+  profileIdentifier,
+  loggedInUserIdentifier,
+}: IGetProfileDataArgs): Promise<IProfileData> => {
+  const profileData = await getMumbleUserByIdentifier(profileIdentifier);
+  const profileFollowers = await getAllFollowers({
+    identifier: profileIdentifier,
+  });
+  const profileFollowees = await getAllFollowees(profileIdentifier);
```

Comment:
if you visit some other user profile (not yours), you need to know the followers of that profile, so that you can list them as user cards. Also you need to know all followers of the logged in account, so that you can follow or unfollow the listed users in this other profile.